internal/keycloakb/archivedbmodule.go

```diff
@@ -45,13 +45,13 @@ func (a *archiveDBModule) StoreUserDetails(ctx context.Context, realm string, us
 	}
 
 	// encrypt the JSON containing the details on the user
-	encryptedData, err := a.cipher.Encrypt(details, []byte(*user.ID))
+	encryptedData, keyId, err := a.cipher.Encrypt(details, []byte(*user.ID))
```

Comment:
I think it would be better to have a separate method to retrieve the currently used keyId. (See comment in common-service)